orangecontrib/text/widgets/owconcordance.py

```diff
@@ -259,17 +255,21 @@ def _set_selection(self, selection):
     def set_corpus(self, data=None):
         self.closeContext()
         self.corpus = data
-        if data is not None and not isinstance(data, Corpus):
-            self.corpus = Corpus.from_table(data.domain, data)
-        self.model.set_corpus(self.corpus)
+        if data is None:    # data removed, clear selection
+            self.selected_rows = []
+
         if not self.is_word_on_input:
             self.word = ""
             self.openContext(self.corpus)
+
+        self.model.set_corpus(self.corpus)
         self.set_word()
 
     @Inputs.query_word
     def set_word_from_input(self, topic):
         self.Warning.multiple_words_on_input.clear()
+        if self.is_word_on_input:   # word changed, clear selection
+            self.selected_rows = []
```

Comment:
Why `if`?

Reply:
So it only resets when we change the input word from one topic to another and not also when we reload the scema and topic is passed on input.

I think that without this when reloading selection would be reset before `handleNewSignals` would have the chance to set it.